modules/pathtracing/glsl/bidirectionalvolumepathtracer.comp

```diff
@@ -0,0 +1,233 @@
+#include "utils/structs.glsl"
+#include "utils/sampler2d.glsl"
+#include "utils/sampler3d.glsl"
+
+#include "utils/classification.glsl"
+#include "utils/compositing.glsl"
+#include "utils/depth.glsl"
+#include "utils/gradients.glsl"
+#include "utils/shading.glsl"
+#include "utils/raycastgeometry.glsl"
+
+#include "random.glsl"
+#include "transmittancemethods.glsl"
+#include "renderingmethods.glsl"
+
+// volume
+uniform sampler3D volume;
+uniform VolumeParameters volumeParameters;
+
+// entry
+uniform sampler2D entryColor;
+uniform sampler2D entryDepth;
+uniform ImageParameters entryParameters;
+
+// exit
+uniform sampler2D exitColor;
+uniform sampler2D exitDepth;
+uniform ImageParameters exitParameters;
+
+// output
+uniform layout(binding=0, rgba8) image2D outportColor;
+writeonly uniform image2D outportDepth;
+writeonly uniform image2D outportPicking;
+uniform ImageParameters outportParameters;
+
+// Uniform properties
+uniform CameraParameters camera;
+uniform VolumeIndicatorParameters positionindicator;
+uniform RaycastingParameters raycaster;
+uniform LightParameters light; //lighting
+uniform sampler2D transferFunction; 
+
+uniform int channel;
+
+uniform float time_ms;
+uniform int iteration;
+
+#define ERT_THRESHOLD 0.99  // threshold for early ray termination
+
+#define PI 3.1415
+
+layout (local_size_x = 16, local_size_y = 16) in;
+
+float calcStep(in float rayLength, in vec3 direction, in float samplingRate, in vec3 dimensions) {
+    float incr = min(rayLength, rayLength / (samplingRate * length(direction * dimensions)));
+    float samples = ceil(rayLength / incr);
+    return rayLength / samples;
+}
+
+vec3 calcCameraDir(in vec3 entryPoint, in vec3 exitPoint, in mat4 textureToWorld) {
```

Comment:
Created new file raycasting.glsl in opengl/glsl/utils/ and added it to CMakeLists, mainting alphabetical order.
`
/*********************************************************************************
 *
 * Inviwo - Interactive Visualization Workshop
 *
 * Copyright (c) 2015-2023 Inviwo Foundation
 * All rights reserved.
 *
 * Redistribution and use in source and binary forms, with or without
 * modification, are permitted provided that the following conditions are met:
 *
 * 1. Redistributions of source code must retain the above copyright notice, this
 * list of conditions and the following disclaimer.
 * 2. Redistributions in binary form must reproduce the above copyright notice,
 * this list of conditions and the following disclaimer in the documentation
 * and/or other materials provided with the distribution.
 *
 * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
 * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
 * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
 * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
 * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
 * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
 * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
 * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
 * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
 * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 *
 *********************************************************************************/

#ifndef IVW_RAYCASTING_GLSL
#define IVW_RAYCASTING_GLSL

vec3 calcCameraDir(in vec3 entryPoint, in vec3 exitPoint, in mat4 textureToWorld) {
    return normalize(
        (textureToWorld * vec4(entryPoint, 1.0) - textureToWorld * vec4(exitPoint, 1.0)).xyz);
}

#endif //IVW_RAYCASTING_GLSL
`
would these changes make the cut